blockmanager.go

```diff
@@ -179,10 +180,7 @@ func (b *blockManager) resetHeaderState(newestHash *chainhash.Hash, newestHeight
 func (b *blockManager) findNextHeaderCheckpoint(height int32) *chaincfg.Checkpoint {
 	// There is no next checkpoint if checkpoints are disabled or there are
 	// none for this current network.
-	if cfg.DisableCheckpoints {
-		return nil
-	}
-	checkpoints := b.server.chainParams.Checkpoints
+	checkpoints := b.chain.Checkpoints()
```

Comment:
I think the check for disable should be kept.  You're making assumptions about the state of chain's checkpoints here that could change in a refactor.  The code is more robust with the check of `cfg.DisableCheckpoints ` (and it keeps the comment accurate).